### PR TITLE
Accurate totals in response metadata

### PR DESCRIPTION
**Pull request by trieloff**
This PR extends #920 with a relatively minor feature that required some bigger refactoring.

When using parameters like `limit` and `offset` to enable pagination, it is not always clear how many results there are. Previously, we'd simply report the number of filtered results as total (or increased by one to indicate that the response set was truncated for some reason – such as response size limits). This is very inaccurate.

With this PR, query authors will have the ability to report accurate total result numbers by turning their SQL into multi-statement files. The final statement will still be the returned data, but if the query is flagged with `# hlx:metadata` then the penultimate statement will take the `total_rows` value of the first returned row and include it in the response metadata.

### CODE REVIEW

**Comment by langswei** [comment]
@trieloff https://helix-pages.anywhere.run/helix-services/run-query@ci5595/multi-results?url=www.hlx.live&domain=www.hlx.live does not load for me when I attempted to verify the output, I think package.json needs to be adjusted to include the sample query.

**Comment by langswei** [comment]
I created a branch `accurate-totals-test`, updated package.json, and ran the query at https://helix-pages.anywhere.run/helix-services/run-query@ci5699/multi-results?url=www.hlx.live&domain=www.hlx.live.  The relevant bit of the result of the query defined at https://github.com/adobe/helix-run-query/blob/accurate-totals-test/src/queries/multi-results.sql is:

`"results":{"limit":1,"offset":0,"total":100,"data":[{"is_admin":false}],"columns":["is_admin"]}`

whereas previous versions would have said `"total":1`.

https://github.com/adobe/helix-run-query/blob/37261d363999ade2dbe6d9c684478b0fcd35de50/src/sendquery.js#L144 The above comment does not feel deterministic but seems to work and I cannot think of a scenario where a count query is going to be slower than the actual query with results.

I approve this PR.  @trieloff to merge upon his return from PTO -- I wasn't sure if appropriate to merge `accurate-totals-test` branch to `accurate-totals` first or simply delete that branch.  Once this PR is merged, combined with discussion at #926 it will help unblock a real-world test for @dbrrt for pagination for the BDF page views query.

**Review by langswei — APPROVED** [review]

File: src/sendquery.js
Comment (dbrrt) [inline src/sendquery.js:147]:
```suggestion
              const [metadataResults] = await metadata.getQueryResults();
              responseMetadata.totalRows = metadataResults[0]?.total_rows;
```

**Review by dbrrt — APPROVED** [review]
LGTM!

**Review by MarquiseRosier — APPROVED** [review]

File: src/sendquery.js
Comment (dbrrt) [reply src/sendquery.js]: @trieloff ?

Comment (dbrrt) [reply src/sendquery.js]: Did you see the typo?

**Pull request merged**
